synth-publish.go

```diff
@@ -220,8 +221,41 @@ func checkPublishingStatus(latest postedData, result chan<- publicationResult, s
 
 	switch resp.StatusCode {
 	case http.StatusOK:
+		cmdR := exec.Command("kubectl", "delete", "cm", "synthetic-image-alarm", "--ignore-not-found")
+		cmdI := exec.Command("kubectl", "delete", "pods", "--selector=job-name=image-trace-job", "--ignore-not-found=true")
+
+		errR := cmdR.Run()
+		if errR != nil {
+			log.Fatalf("cmdR.Run() failed with %s\n", errR)
+		}
+
+		errI := cmdI.Run()
+		if errI != nil {
+			log.Fatalf("cmdR.Run() failed with %s\n", errI)
+		}
 	case http.StatusNotFound:
 		handlePublishingErr(result, latest.tid, latest.time, "Image not found. Response status code: 404.")
+		cmd1 := exec.Command("kubectl", "delete", "jobs", "image-trace-job", "--ignore-not-found")
```

Comment:
Just a recommendation, since in the script there is already a functions i'd use a func for each case: 

`func k8sNotFound() error {

	var err error
	cmd1 := exec.Command("kubectl", "delete", "jobs", "image-trace-job", "--ignore-not-found")
	cmd2 := exec.Command("kubectl", "delete", "configmap", "synthetic-tid")
	cmd3 := exec.Command("kubectl", "create", "configmap", "synthetic-tid", "--from-literal=TID="+latest.tid)
	cmd4 := exec.Command("kubectl", "create", "job", "--from=cronjob/image-trace", "image-trace-job")
	
	err = cmd1.Run()
	if err1 != nil {
		return fmt.Errorf("cmd1.Run() failed with %s\n", err.Error())
	}
	err = cmd2.Run()
	if err != nil {
		return fmt.Errorf("cmd2.Run() failed with %s\n", err.Error())
	}
	err = cmd3.Run()
	if err != nil {
		return fmt.Errorf("cmd3.Run() failed with %s\n", err.Error())
	}
	err = cmd4.Run()
	if err != nil {
		return fmt.Errorf("cmd4.Run() failed with %s\n", err.Error())
	}
}

if err := k8sNotFound(); err != nil {
	log.Fatal(err.Error())
}`

Reply:
There is "latest.tid" here which is only valid in checkPublishingStatus function so I've decided to not use such function